### PR TITLE
Print stdout and stderr when pod validation fails

### LINKED ISSUE #370: Improve control plane validation play
We need to be able to debug when failures occur on validating the control plane.

Docker logs and the kubelet logs would be a good place to start.

### PR DESCRIPTION
**Pull request by dkoshkin**
Closes #370 

Validation will now run on each node
On failure the docker logs will be collected and printed in the cli output

I've tried using `handlers` and `rescue` block but neither worked; there is no way to avoid seeing the task name with ` [SKIPPED]` 

```
Start Kubernetes Controller Manager Finished
Validate Kubernetes Control Plane is Running
- Running task: wait until kube-apiserver-master001 pod has the desired version
  master001                                                                     [OK]
  master002                                                                     [OK]
- Running task: wait until kube-apiserver-master001 pod is running
 master001 Retrying: wait until kube-apiserver-master001 pod is running (1/11 attempts)
 master002 Retrying: wait until kube-apiserver-master001 pod is running (1/11 attempts)
 master001 Retrying: wait until kube-apiserver-master001 pod is running (2/11 attempts)
 master002 Retrying: wait until kube-apiserver-master001 pod is running (2/11 attempts)
 master002 Retrying: wait until kube-apiserver-master001 pod is running (3/11 attempts)
 master001 Retrying: wait until kube-apiserver-master001 pod is running (3/11 attempts)
 master002 Retrying: wait until kube-apiserver-master001 pod is running (4/11 attempts)
 master001 Retrying: wait until kube-apiserver-master001 pod is running (4/11 attempts)
 master002 Retrying: wait until kube-apiserver-master001 pod is running (5/11 attempts)
 master001 Retrying: wait until kube-apiserver-master001 pod is running (5/11 attempts)
 master002 Retrying: wait until kube-apiserver-master001 pod is running (6/11 attempts)
 master001 Retrying: wait until kube-apiserver-master001 pod is running (6/11 attempts)
 master002 Retrying: wait until kube-apiserver-master001 pod is running (7/11 attempts)
 master001 Retrying: wait until kube-apiserver-master001 pod is running (7/11 attempts)
 master002 Retrying: wait until kube-apiserver-master001 pod is running (8/11 attempts)
 master001 Retrying: wait until kube-apiserver-master001 pod is running (8/11 attempts)
 master002 Retrying: wait until kube-apiserver-master001 pod is running (9/11 attempts)
 master001 Retrying: wait until kube-apiserver-master001 pod is running (9/11 attempts)
 master002 Retrying: wait until kube-apiserver-master001 pod is running (10/11 attempts)
 master001 Retrying: wait until kube-apiserver-master001 pod is running (10/11 attempts)
  master002                                                                     [OK]
  master001                                                                     [OK]
- Running task: get docker container ID for kube-apiserver-master001
  master001                                                                     [OK]
  master002                                                                     [OK]
- Running task: get docker logs for kube-apiserver-master001
  master001                                                                     [ERROR]
---- STDERR ----
I0307 20:13:00.650039       1 config.go:527] Will report 192.168.42.3 as public IP address.
I0307 20:13:00.650437       1 server.go:148] Initalizing deserialization cache size based on 0MB limit
I0307 20:13:00.652168       1 server.go:342] Initalizing cache sizes based on 0MB limit
I0307 20:13:00.701923       1 master.go:310] Enabling API group "apps".
I0307 20:13:00.702177       1 master.go:310] Enabling API group "authentication.k8s.io".
I0307 20:13:00.702234       1 master.go:310] Enabling API group "authorization.k8s.io".
I0307 20:13:00.703525       1 master.go:310] Enabling API group "autoscaling".
I0307 20:13:00.704509       1 master.go:310] Enabling API group "batch".
I0307 20:13:00.705464       1 master.go:310] Enabling API group "certificates.k8s.io".
I0307 20:13:00.713840       1 master.go:310] Enabling API group "extensions".
I0307 20:13:00.714567       1 master.go:310] Enabling API group "policy".
I0307 20:13:00.720747       1 master.go:310] Enabling API group "rbac.authorization.k8s.io".
I0307 20:13:00.725091       1 master.go:310] Enabling API group "storage.k8s.io".
[restful] 2017/03/07 20:13:00 log.go:30: [restful/swagger] listing is available at https://192.168.42.3:6443/swaggerapi/
[restful] 2017/03/07 20:13:00 log.go:30: [restful/swagger] https://192.168.42.3:6443/swaggerui/ is mapped to folder /swagger-ui/
I0307 20:13:00.932549       1 serve.go:88] Serving securely on 0.0.0.0:6443
F0307 20:13:00.932905       1 genericapiserver.go:195] failed to listen on 0.0.0.0:6443: listen tcp 0.0.0.0:6443: bind: address already in use
---------------
  master002                                                                     [ERROR]
---- STDERR ----
I0307 20:12:58.029456       1 config.go:527] Will report 192.168.42.5 as public IP address.
I0307 20:12:58.029513       1 server.go:148] Initalizing deserialization cache size based on 0MB limit
I0307 20:12:58.031197       1 server.go:342] Initalizing cache sizes based on 0MB limit
I0307 20:12:58.086231       1 master.go:310] Enabling API group "apps".
I0307 20:12:58.087138       1 master.go:310] Enabling API group "authentication.k8s.io".
I0307 20:12:58.087193       1 master.go:310] Enabling API group "authorization.k8s.io".
I0307 20:12:58.088636       1 master.go:310] Enabling API group "autoscaling".
I0307 20:12:58.089870       1 master.go:310] Enabling API group "batch".
I0307 20:12:58.090552       1 master.go:310] Enabling API group "certificates.k8s.io".
I0307 20:12:58.099300       1 master.go:310] Enabling API group "extensions".
I0307 20:12:58.100042       1 master.go:310] Enabling API group "policy".
I0307 20:12:58.112186       1 master.go:310] Enabling API group "rbac.authorization.k8s.io".
I0307 20:12:58.113423       1 master.go:310] Enabling API group "storage.k8s.io".
[restful] 2017/03/07 20:12:58 log.go:30: [restful/swagger] listing is available at https://192.168.42.5:6443/swaggerapi/
[restful] 2017/03/07 20:12:58 log.go:30: [restful/swagger] https://192.168.42.5:6443/swaggerui/ is mapped to folder /swagger-ui/
I0307 20:12:58.297042       1 serve.go:88] Serving securely on 0.0.0.0:6443
F0307 20:12:58.297227       1 genericapiserver.go:195] failed to listen on 0.0.0.0:6443: listen tcp 0.0.0.0:6443: bind: address already in use
---------------
Validate Kubernetes Control Plane is Running Finished
error installing: error running playbook: error running ansible: exit status 2
```

When the pod is running the tasks will be skipped and the output will look like:
```                                         
Validate Kubernetes Control Plane is Running
- Running task: wait until kube-apiserver-master001 pod has the desired version
  master001                                                                     [OK]
  master002                                                                     [OK]
- Running task: wait until kube-apiserver-master001 pod is running
  master002                                                                     [OK]
  master001                                                                     [OK]
- Running task: get docker container ID for kube-apiserver-master001
  master001                                                                     [SKIPPED]
  master002                                                                     [SKIPPED]
- Running task: get docker logs for kube-apiserver-master001
  master001                                                                     [SKIPPED]
  master002                                                                     [SKIPPED]
- Running task: fail if kube-apiserver-master001 pod is not running
  master001                                                                     [SKIPPED]
  master002                                                                     [SKIPPED]
- Running task: wait until kube-scheduler-master001 pod has the desired version
  master001                                                                     [OK]
  master002                                                                     [OK]
- Running task: wait until kube-scheduler-master001 pod is running
  master001                                                                     [OK]
  master002                                                                     [OK]
- Running task: get docker container ID for kube-scheduler-master001
  master001                                                                     [SKIPPED]
  master002                                                                     [SKIPPED]
- Running task: get docker logs for kube-scheduler-master001
  master001                                                                     [SKIPPED]
  master002                                                                     [SKIPPED]
- Running task: fail if kube-scheduler-master001 pod is not running
  master001                                                                     [SKIPPED]
  master002                                                                     [SKIPPED]
- Running task: wait until kube-controller-manager-master001 pod has the desired version
  master001                                                                     [OK]
  master002                                                                     [OK]
- Running task: wait until kube-controller-manager-master001 pod is running
  master002                                                                     [OK]
  master001                                                                     [OK]
- Running task: get docker container ID for kube-controller-manager-master001
  master001                                                                     [SKIPPED]
  master002                                                                     [SKIPPED]
- Running task: get docker logs for kube-controller-manager-master001
  master001                                                                     [SKIPPED]
  master002                                                                     [SKIPPED]
- Running task: fail if kube-controller-manager-master001 pod is not running
  master001                                                                     [SKIPPED]
  master002                                                                     [SKIPPED]                     
```

### CODE REVIEW

File: ansible/kubernetes.yaml [hunk 1]
```diff
@@ -22,7 +22,7 @@
   - include: _kube-scheduler.yaml
   - include: _kube-controller-manager.yaml
   # validating has a dependecy on the API server for the static pods
-  - include: _validate-control-plane.yaml
+  - include: _validate-control-plane-node.yaml
```
Comment (alexbrand) [inline ansible/kubernetes.yaml]: Do we need to run validation multiple times? I think this change will make it run on all the masters?

I guess it doesn't hurt.

Comment (dkoshkin) [inline ansible/kubernetes.yaml]: It will run the validation only for the pods on that node, this allowed me to get to the docker logs

Comment (alexbrand) [inline ansible/kubernetes.yaml]: Oh nice!

**Comment by alexbrand** [comment]
LGTM

**Pull request merged**